### PR TITLE
[3D Object Detection] Waymo Evaluation Callback

### PR DESCRIPTION
**Pull request by ianstenbit**
See test for example usage.

I suspect that the input/output shapes may need to be tweaked once we have the pipeline set up, but the core of it is here.

The tricky part, as it turns out, is that WOD's metric evaluator can't handle batched inputs, so we have to manually un-batch everything if we want to avoid calling the model 1 input at a time.

Update: this now filters out boxes with class=-1 (to handle padding ragged to dense with -1s)

### CODE REVIEW

**Comment by ianstenbit** [comment]
@tanzhenyu @Guowang @lengzq I've updated the metrics callback to accept the output shape that we expect from the model (separate boxes and non-argmaxed class preds)

**Review by tanzhenyu — APPROVED** [review]

File: keras_cv/callbacks/__init__.py [hunk 1]
```diff
@@ -12,3 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from keras_cv.callbacks.pycoco_callback import PyCOCOCallback
+
+try:
+    from keras_cv.callbacks.waymo_evaluation_callback import WaymoEvaluationCallback
+except ImportError:
+    pass
```
[the remark this answers is not in the document]
Comment (ianstenbit) [reply keras_cv/callbacks/__init__.py]: Added an import warning here (same we have for PyCocotools)

**Comment by ianstenbit** [comment]
GPU test failure is unreleated to this PR. I am going to open a separate PR to resolve that.

**Pull request merged**